livre/lib/data/BoundingAxis.cpp

```diff
@@ -0,0 +1,374 @@
+
+    glGenBuffers( 1, &_colorBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _colorBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLuint) * colors.size(),
+                  &colors[0], GL_STATIC_DRAW );
+
+    glGenBuffers( 1, &_typeBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLubyte) * types.size(),
+                  &types[0], GL_STATIC_DRAW );
+}
+
+void BoundingAxis::draw()
+{
+    glEnableVertexAttribArray( 0 );
+    glBindBuffer( GL_ARRAY_BUFFER, _vertexeBufferId );
+    glVertexAttribPointer( 0, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 1 );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId );
+    glVertexAttribPointer( 1, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 2 );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId2 );
+    glVertexAttribPointer( 2, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 3 );
+    glBindBuffer( GL_ARRAY_BUFFER, _colorBufferId );
+    glVertexAttribPointer( 3, 4, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 4 );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glVertexAttribIPointer( 4, 1, GL_UNSIGNED_BYTE, 0, 0 );
+
+    glDrawArrays( GL_LINES, 0, nVertices );
+
+    glDisableVertexAttribArray( 0 );
+    glDisableVertexAttribArray( 1 );
+    glDisableVertexAttribArray( 2 );
+    glDisableVertexAttribArray( 3 );
+}
+
+void BoundingAxis::_cleanGPUMemory()
+{
+    glDeleteBuffers( 1, &_vertexeBufferId );
+    glDeleteBuffers( 1, &_normalBufferId );
+    glDeleteBuffers( 1, &_normalBufferId2 );
+    glDeleteBuffers( 1, &_colorBufferId );
+    glDeleteBuffers( 1, &_typeBufferId );
+}
+
+void BoundingAxis::_createTicks( const VolumeInformation& volInfo, Floats& vertices,
+                                 Floats& normals, Floats& normals2, Floats& colors, UInt8s& types )
+{
+    const size_t longestDimCoordIndex = volInfo.worldSize.find_max_index();
+
+    const Vector2f range = _computeRange( volInfo, longestDimCoordIndex );
+
+    const uint32_t maxTickNumber = 50;
+    const float tickSizes[4] = { 1, 2, 5, 10};
+    float tickSize = 0;
+
+    const float rangeLenght = range.y() - range.x();
+    const float rawTickLenght = rangeLenght / maxTickNumber;
+
+    const float factor = std::pow( 10, std::floor( std::log10( rawTickLenght )));
+    const float normalizedTickLenght = rawTickLenght / factor;
+
+    for( uint32_t i = 0; i < 4; ++i )
+    {
+        if( normalizedTickLenght <= tickSizes[i] )
+        {
+            tickSize = tickSizes[i] * factor;
+            break;
+        }
+    }
+
+    const float x = volInfo.worldSize.x() * 0.5f;
+    const float y = volInfo.worldSize.y() * 0.5f;
+    const float z = volInfo.worldSize.z() * 0.5f;
+
+    _createLocalTicks( volInfo, { -x, -y, z }, { x, -y, z }, { 0, 1, 0 }, { 0, 0, -1 },
+                       false, tickSize, factor, vertices, normals, normals2, colors, types );
```

Comment:
Why ? Maybe I don't undestand the comment because for me if I add a struct it will only increase the size of code.